cupyx/scipy/sparse/index.py

```diff
@@ -0,0 +1,419 @@
+"""Indexing mixin for sparse matrix classes.
+"""
+from .sputils import isintlike
+
+from numpy import integer
+
+import cupy
+from cupy import core
+
+
+INT_TYPES = (int, integer)
+
+
+def _csr_column_inv_idx(idxs):
+    """Construct an inverted index, mapping the indices
+    of the given array to the their values
+
+    Args
+        idxs : array of indices to invert
+        tpb : threads per block for underlying kernel
+
+    Returns
+        idxs_adj : inverted indices where idxs_adj[idxs[i]] = i
+    """
+    max_idx = idxs.max().item()
+    idxs_adj = cupy.zeros(max_idx + 1, dtype=idxs.dtype)
```

Comment:
no need to launch the zero-out kernel:
```suggestion
    idxs_adj = cupy.empty(max_idx + 1, dtype=idxs.dtype)
```